control-plane/api-gateway/binding/annotations.go

```diff
@@ -25,7 +21,7 @@ func serializeGatewayClassConfig(gw *gwv1beta1.Gateway, gwcc *v1alpha1.GatewayCl
 		gw.Annotations = make(map[string]string)
 	}
 
-	if annotatedConfig, ok := gw.Annotations[annotationConfigKey]; ok {
+	if annotatedConfig, ok := gw.Annotations[common.AnnotationGatewayClassConfig]; ok {
```

Comment:
out of curiosity why create the `common` package and move a bunch of this into it?

Reply:
Part of it is because there was a lot of code duplication, especially of common helpers across the code -- there are multiple places where something like a `pointerTo`-like method is introduced, where constants were across multiple packages, etc. In addition, due to circular reference issues, the top-level package started to become a larger dumping ground for some of the helpers/structs that were being used across other subpackages. Rather than have a giant top-level package imported everywhere that was the common structs/helpers/constants we use across everything else, figured it'd make more sense to just make it explicit and call it `common`

Reply:
Yeah, I normally don't like having a "common" or "utils" package, but here it definitely makes sense.